core/src/main/scala/cats/std/list.scala

```diff
@@ -70,6 +71,11 @@ trait ListInstances extends ListInstances1 {
 
   implicit def listAlgebra[A]: Monoid[List[A]] = new ListMonoid[A]
   implicit def listOrder[A: Order]: Order[List[A]] = new ListOrder[A]
+
+  implicit def listShow[A:Show]: Show[List[A]] =
+    new Show[List[A]] {
+      def show(fa: List[A]): String = s"List(${fa.map(_.show).mkString(", ")})"
```

Comment:
If you're already using `mkString`: `fa.map(_.show).mkString("List(", ", ", ")")`